File: Definitions/Game_Definitions/classes/Fuse.lua
---@meta
---@diagnostic disable

---@class Fuse : InteractiveMasterDevice
Fuse = {}

---@return Fuse
function Fuse.new() return end

---@param props table
---@return Fuse
function Fuse.new(props) return end

---@param ri entEntityRequestComponentsInterface
---@return Bool
function Fuse:OnRequestComponents(ri) return end

---@param ri entEntityResolveComponentsInterface
---@return Bool
function Fuse:OnTakeControl(ri) return end

---@return FuseController
function Fuse:GetController() return end

---@return FocusForcedHighlightData
function Fuse:GetDefaultHighlight() return end

---@return FuseControllerPS
function Fuse:GetDevicePS() return end

---@return Bool
function Fuse:IsGameplayRelevant() return end

---@return Bool
function Fuse:ShouldSendGameAttachedEventToPS() return end

File: DecompiledGameScripts/cyberpunk/devices/masters/fuse.swift

public class Fuse extends InteractiveMasterDevice {

  protected cb func OnRequestComponents(ri: EntityRequestComponentsInterface) -> Bool {
    EntityRequestComponentsInterface.RequestComponent(ri, n"DeviceTimetable", n"DeviceTimetable", true);
    super.OnRequestComponents(ri);
  }

  protected cb func OnTakeControl(ri: EntityResolveComponentsInterface) -> Bool {
    this.m_controller = EntityResolveComponentsInterface.GetComponent(ri, n"controller") as FuseController;
  }

  protected const func GetController() -> ref<FuseController> {
    return this.m_controller as FuseController;
  }

  public const func GetDevicePS() -> ref<FuseControllerPS> {
    return this.GetController().GetPS();
  }

  public const func GetDefaultHighlight() -> ref<FocusForcedHighlightData> {
    return null;
  }

  public const func IsGameplayRelevant() -> Bool {
    return false;
  }

  public const func ShouldSendGameAttachedEventToPS() -> Bool {
    return false;
  }
}
